src/ui/toolbar/MainToolBar.cc

```diff
@@ -258,14 +264,15 @@ void MainToolBar::_setActiveUAS(UASInterface* active)
     // If switching the UAS, disconnect the existing one.
     if (_mav)
     {
-        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageReceived,  this, &MainToolBar::_handleTextMessage);
-        disconnect(_mav, &UASInterface::heartbeatTimeout,                                   this, &MainToolBar::_heartbeatTimeout);
-        disconnect(_mav, &UASInterface::batteryChanged,                                     this, &MainToolBar::_updateBatteryRemaining);
-        disconnect(_mav, &UASInterface::modeChanged,                                        this, &MainToolBar::_updateMode);
-        disconnect(_mav, &UASInterface::nameChanged,                                        this, &MainToolBar::_updateName);
-        disconnect(_mav, &UASInterface::systemTypeSet,                                      this, &MainToolBar::_setSystemType);
-        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),              this, SLOT(_updateState(UASInterface*,QString,QString)));
-        disconnect(_mav, SIGNAL(armingChanged(bool)),                                       this, SLOT(_updateArmingState(bool)));
+        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageCountChanged,  this, &MainToolBar::_handleTextMessage);
+        disconnect(_mav, &UASInterface::heartbeatTimeout,                                       this, &MainToolBar::_heartbeatTimeout);
+        disconnect(_mav, &UASInterface::batteryChanged,                                         this, &MainToolBar::_updateBatteryRemaining);
+        disconnect(_mav, &UASInterface::modeChanged,                                            this, &MainToolBar::_updateMode);
+        disconnect(_mav, &UASInterface::nameChanged,                                            this, &MainToolBar::_updateName);
+        disconnect(_mav, &UASInterface::systemTypeSet,                                          this, &MainToolBar::_setSystemType);
+        disconnect(_mav, &UASInterface::localizationChanged,                                    this, &MainToolBar::_setSatLoc);
+        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),                  this, SLOT(_updateState(UASInterface*,QString,QString)));
```

Comment:
If you mean ```statusChanged()```, I didn't because it is an overloaded signal (really don't like those).